lib/src/marker_widget.dart

```diff
@@ -19,17 +19,16 @@ class MarkerWidget extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final m = marker.builder(context);
-    return buildOnHover
-        ? GestureDetector(
-            behavior: HitTestBehavior.opaque,
-            onTap: hoverOnTap ?? onTap,
-            child: onHover != null
-                ? MouseRegion(
-                    onEnter: (_) => onHover!(true),
-                    onExit: (_) => onHover!(false),
-                    child: m,
-                  )
-                : m)
-        : m;
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      onTap: hoverOnTap ?? onTap,
+      child: onHover != null
+          ? MouseRegion(
+              onEnter: (_) => onHover!(true),
+              onExit: (_) => onHover!(false),
+              child: m,
+            )
+          : m,
+    );
```

Comment:
This will be a proper fix:
```suggestion
    return GestureDetector(
        behavior: HitTestBehavior.opaque,
        onTap: buildOnHover && hoverOnTap != null ? hoverOnTap : onTap,
        child: buildOnHover && onHover != null
            ? MouseRegion(
                onEnter: (_) => onHover!(true),
                onExit: (_) => onHover!(false),
                child: m,
              )
            : m);
```